ts/features/wallet/bancomatpay/component/__test__/BPayWalletPreview.test.tsx

```diff
@@ -0,0 +1,132 @@
+    const myspy = jest.spyOn(hooks, "useImageResize").mockReturnValue(none);
+    const component = getComponent(aBPay, store);
+    const bankLogo = component.queryByTestId("bankLogoFallback");
+
+    expect(bankLogo).not.toBeNull();
+    expect(bankLogo).toHaveTextContent("BANCOMAT Pay");
+    expect(myspy).toHaveBeenCalledTimes(1);
+  });
+  it("should show the generic bankName string if there isn't the abiInfo", () => {
+    jest.spyOn(hooks, "useImageResize").mockReturnValue(none);
+    const bankName = "INTESA SANPAOLO - S.P.A.";
+    const component = getComponent(
+      {
+        ...aBPay,
+        info: { ...aBPay.info, bankName }
+      },
+      store
+    );
+    const bankLogo = component.queryByTestId("bankLogoFallback");
+
+    expect(bankLogo).not.toBeNull();
+    expect(bankLogo).toHaveTextContent(bankName);
+  });
+
+  it("should show the bankName from the abiInfo if there isn't the logoUrl", () => {
+    jest.spyOn(hooks, "useImageResize").mockReturnValue(none);
+    const infobankName = "a different bank name";
+    const abiInfoBankName = "INTESA SANPAOLO - S.P.A.";
+    const component = getComponent(
+      {
+        ...aBPay,
+        info: { ...aBPay.info, bankName: infobankName },
+        abiInfo: {
+          abi: "03069",
+          name: abiInfoBankName
+        }
+      },
+      store
+    );
+    const bankLogo = component.queryByTestId("bankLogoFallback");
+
+    expect(bankLogo).not.toBeNull();
+    expect(bankLogo).toHaveTextContent(abiInfoBankName);
+  });
+
+  it("should show the logo image if there is the abiInfo logoUrl", () => {
+    jest.spyOn(hooks, "useImageResize").mockReturnValue(some([15, 15]));
+
+    const infobankName = "a different bank name";
+    const abiInfoBankName = "INTESA SANPAOLO - S.P.A.";
+    const logoUrl = "http://127.0.0.1:3000/static_contents/logos/abi/03069.png";
+    const component = getComponent(
+      {
+        ...aBPay,
+        info: { ...aBPay.info, bankName: infobankName },
+        abiInfo: {
+          abi: "03069",
+          name: abiInfoBankName,
+          logoUrl
+        }
+      },
+      store
+    );
+    const bankLogo = component.queryByTestId("bankLogo");
+
+    expect(bankLogo).not.toBeNull();
+    expect(bankLogo).toHaveProp("source", { uri: logoUrl });
+  });
+
+  it("should call navigateToBPayDetails when press on it", () => {
+    jest.spyOn(hooks, "useImageResize").mockReturnValue(none);
+    const component = getComponent(aBPay, store);
+    const cardComponent = component.queryByTestId("cardPreview");
+    const expectedPayload = {
+      type: NavigationActions.NAVIGATE,
+      routeName: ROUTES.WALLET_BPAY_DETAIL,
+      params: { bPay: aBPay }
+    };
+    if (cardComponent) {
+      fireEvent.press(cardComponent);
+      expect(store.getActions()).toEqual([expectedPayload]);
+    }
+  });
+});
+
+const getComponent = (bPay: BPayPaymentMethod, store: any) =>
```

Comment:
```suggestion
const getComponent = (bPay: BPayPaymentMethod, store: Store<unknown>) =>
```
The same as before, just to avoid `any` type.